src/Raven.Server/Web/System/AdminCpuCreditsHandler.cs

```diff
@@ -26,6 +28,29 @@ public Task UpdateCpuCredits()
             }
         }
 
+        [RavenAction("/admin/cpu-credits/sync", "POST", AuthorizationStatus.ClusterAdmin)]
+        public Task SyncCpuCredits()
+        {
+            try
+            {
+                Server.UpdateCpuCreditsFromExec();
```

Comment:
This is something that we really want to avoid. We are turning everything to async, so let's make this async and wait in the long running thread, instead.

Reply:
Yes of course.